### PR TITLE
Extract alamofire requests add custom chain 

### PR DESCRIPTION
**Pull request by oa-s**

### CODE REVIEW

**Comment by oa-s** [comment]
intermediate part of request standartization

File: AlphaWallet/ActiveWalletCoordinator.swift [hunk 1]
```diff
@@ -376,21 +376,43 @@ class ActiveWalletCoordinator: NSObject, Coordinator, DappRequestHandlerDelegate
     }
 
     private func createActivityCoordinator(activitiesService: ActivitiesServiceType) -> ActivitiesCoordinator {
-        let coordinator = ActivitiesCoordinator(analytics: analytics, sessions: sessionsProvider.activeSessions, activitiesService: activitiesService, keystore: keystore, wallet: wallet, assetDefinitionStore: assetDefinitionStore)
+        let coordinator = ActivitiesCoordinator(
```
Comment (hboon) [inline AlphaWallet/ActiveWalletCoordinator.swift:379]: One long-line and let it soft-wrap instead?

File: AlphaWallet/ActiveWalletCoordinator.swift [hunk 2]
```diff
@@ -624,14 +649,33 @@ extension ActiveWalletCoordinator: SelectServiceToBuyCryptoCoordinatorDelegate {
...
+            inViewController: presentationViewController)
+
         coordinator.delegate = dappRequestHandler
         dappRequestHandler.addCoordinator(coordinator)
         coordinator.start()
     }
 
     func requestAddCustomChain(server: RPCServer, callbackId: SwitchCustomChainCallbackId, customChain: WalletAddEthereumChainObject) {
-        let coordinator = DappRequestSwitchCustomChainCoordinator(config: config, server: server, callbackId: callbackId, customChain: customChain, restartQueue: restartQueue, analytics: analytics, currentUrl: nil, inViewController: presentationViewController)
+        let coordinator = DappRequestSwitchCustomChainCoordinator(
```
Comment (hboon) [inline AlphaWallet/ActiveWalletCoordinator.swift:668]: Some of these seems to be formatting changes?

**Comment by hboon** [comment]
But maybe better to get the intermediate PRs/commits merged in first, if that's possible? Less management work for you right?

**Comment by hboon** [comment]
Also maybe:

> Extract alamofire requests add custom chain #5968

to:

> Extract alamofire requests for add custom chain functionality #5968

?

File: modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Models/AddCustomChain.swift [hunk 3]
```diff
@@ -22,7 +28,7 @@ private enum ResolveExplorerApiHostnameError: Error {
 public protocol AddCustomChainDelegate: AnyObject {
     func notifyAddCustomChainQueuedSuccessfully(in addCustomChain: AddCustomChain)
     func notifyAddCustomChainFailed(error: AddCustomChainError, in addCustomChain: AddCustomChain)
-    func notifyAddExplorerApiHostnameFailure(customChain: WalletAddEthereumChainObject, chainId: Int) -> Promise<Bool>
+    func notifyAddExplorerApiHostnameFailure(customChain: WalletAddEthereumChainObject, chainId: Int) -> AnyPublisher<Bool, Never>
```
[the remark this answers is not in the document]
Comment (oa-s) [reply modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Models/AddCustomChain.swift:31]: all network call will be migrated from promises to publishers, (that easier to make retrying logic, can request cancellation), so when we update all chains with publisher, we would be able to cancel all chain operations

**Pull request merged**